Archipelago.RiskOfRain2/Handlers/ScavengerSpawnBoosterHandler.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using R2API;
+using R2API.Utils;
+using RoR2;
+
+namespace Archipelago.RiskOfRain2.Handlers;
+
+public class ScavengerSpawnBoosterHandler : IHandler
+{
+    private readonly LocationHandler _locationHandler;
+
+    internal ScavengerSpawnBoosterHandler(LocationHandler locationHandler)
+    {
+        _locationHandler = locationHandler;
+    }
+
+    public void Hook()
+    {
+        DirectorAPI.MonsterActions += OnMonsterActions;
+    }
+
+    private void OnMonsterActions(DccsPool dccsPool, List<DirectorAPI.DirectorCardHolder> directorCardHolders,
+        DirectorAPI.StageInfo stageInfo)
+    {
+        var stageInt = GetStageInt(stageInfo.stage);
+        var info = _locationHandler.GetLocationsForEnvironment(stageInt)!;
+        var scavengersRemaining = info[LocationHandler.LocationTypes.scavenger];
+        Log.LogInfo($"OnMonsterActions, scavengers remaining : {scavengersRemaining}");
+        if (scavengersRemaining > 0 && info.total() == scavengersRemaining)
+        {
+            var scavengerCardHolder = directorCardHolders
+                .Single(s => s.IsMonster
+                             && s.Card is { spawnCard: CharacterSpawnCard csc }
+                             && csc.name.ToLower() == DirectorAPI.Helpers.MonsterNames.Scavenger.ToLower()
+                );
+            
+            Log.LogInfo($"Card found : {scavengerCardHolder}");
+            Log.LogInfo($"Original cost : {scavengerCardHolder.Card.spawnCard.directorCreditCost}");
+            scavengerCardHolder.Card.spawnCard.directorCreditCost = 500;
+            //todo restore cost when exiting game or changing level, this edit is permanent
+            Log.LogInfo($"New cost : {scavengerCardHolder.Card.cost}");
+        }
+    }
+
+    private int GetStageInt(DirectorAPI.Stage stageInfoStage)
```

Comment:
For some reason DirectorAPI.Stage doesn't have the two alternate stages aka Titanic Plains (2) (golemplains2) and Distant Roost (2) (blackbeach2)
I added a new commit https://github.com/kindasneaki/Archipelago.RiskOfRain2/commit/a2a2500a2aed6b0a4b1a6632d6e58ce0a3304935 to get the current scene instead, and get the sceneDefIndex from that instead of needing the switch to get the index.